docs/pages/zencode-scenarios-qp.md

```diff
@@ -195,7 +195,89 @@ The result should look like:
 
 obviously this secret will be equal to Bob's secret.
 
+# Benchmark
 
+In the following we will compare the Quantum-Proof algorithms with the most common Elliptic-Curve algorithms (ECDSA, ECDH). All the results in the following are generated by the script [benchmark_sig.sh](https://github.com/dyne/Zenroom/blob/master/test/zencode_qp/benchmark_sig.sh) and [benchmark_enc.sh](https://github.com/dyne/Zenroom/blob/master/test/zencode_qp/benchmark_enc.sh).
+
+## Key length
+
+As first parameter we will look at the private and public keys length (measured in bytes).
+
+| Sizes      | private key | public key  |
+|------------|-------------|-------------|
+| ECDSA/ECDH | 32          | 65          |
+| Dilithium  | 2528        | 1312        |
+| Kyber      | 1632        | 800         |
+| NTRUP      | 1763        | 1158        |
+
+As we can see they Quantum-Proof keys are much longer. Will this affect the speed of the algorithms?
+
+## Signature
+
+We can divide the signature in four main parts: the generation of the *private key*, the generation of the *public key*, the *signature* and the *verification*. As can be seen from the tables below the Dilithium *key generation* is **not affected at all from the key length**, indeed both the key generation time (computed in microsecond) and memory (computed in Kibibyte) are not very different between Dilithium and ECDSA.
+
+| Algorithm | Key         | Time       | Memory |
+|-----------|-------------|------------|--------|
+| ECDSA     | private key | 13678,5971 | 617    |
+|           | public key  | 15226,2096 | 615    |
+| Dilithium | private key | 13485,2965 | 650    |
+|           | public key  | 15343,0342 | 637    |
+
+Regarding the *signature* and *verification* algorithms the test was performed on different message lengths: 100, 500, 1000, 2.500, 5.000, 7.500 and 10.000 bytes. The **results are amazing**: the time and memory consumed by the two algorithms are really close to each other. Their are shown in the following tables where time is measured in μs (microsecond) and the memory in KiB (Kibibyte).
+
+- Signature:
+| length | ECDSA Time | Dilithium Time | ECDSA Memory | Dilithium Memory  |
+|--------|------------|----------------|--------------|-------------------|
+| 100    | 16081,6516 | 16836,6952     | 622          | 643               |
+| 500    | 16231,4911 | 17385,6549     | 625          | 646               |
+| 1000   | 16410,2233 | 16711,5771     | 629          | 650               |
+| 2500   | 16900,9023 | 17178,8706     | 641          | 662               |
+| 5000   | 17852,7499 | 18490,5152     | 660          | 681               |
+| 7500   | 18810,2357 | 19303,2178     | 680          | 701               |
+| 10000  | 19812,6163 | 20748,2910     | 699          | 721               |
+
+In order to have a better view of the time consumed by the signature, you can have a look at the following graph:
+![](../_media/images/qp_benchmark_signature.png)
```

Comment:
Maybe we could put the ticks on the x axis for multiple of 1024bytes of 2048bytes, such that they are aligned with the points on the graph. For matplotlib there is the `pyplot.xticks` function, what do you think @matteo-cristino ?

Reply:
Yes, sure thing! The points on the graph are set at 100, 500, 1000, 2500, 5000, 7500 and 10000. Maybe it could be better to use them, what do you think?

Reply:
oh I thought they were powers of two, then leave as they are